src/external_cli/rustup.rs

```diff
@@ -0,0 +1,55 @@
+//! Utilities for the `rustup` CLI tool.
+
+#![expect(dead_code, reason = "Will be used for build/run commands")]
+
+use std::process::{exit, Command};
+
+use dialoguer::Confirm;
+
+const PROGRAM: &str = "rustup";
+
+/// Given a target triple, determine if it is already installed.
+fn is_target_installed(target: &str) -> bool {
+    let output = Command::new(PROGRAM).arg("target").arg("list").output();
+
+    // Check if the target list has an entry like this:
+    // <target_triple> (installed)
+    if let Ok(output) = output {
+        if let Ok(list) = String::from_utf8(output.stdout) {
+            for line in list.lines() {
+                if line.contains(target) && line.contains("(installed)") {
+                    return true;
+                }
+            }
+        }
+    }
+
+    false
+}
+
+/// Install a compilation target, if it is not already installed.
+pub(crate) fn install_target_if_needed(target: &str, ask_user: bool) -> anyhow::Result<()> {
```

Comment:
Not a fan of the boolean parameter, but I don't have a better idea right now.
Maybe we can find a nice name? `ensure_target_by_asking`?

Reply:
Honestly, it's not even used yet so I'll just remove it for now.

In the future, an enum which can confirm without asking, deny without asking or ask the user is probably the way to go.